Add notes for betas for Chocolatey components3rd beta release notes (#701)
chocolatey/docs: input/en-us/licensed-extension/release-notes.md

### Bug Fixes

- Fix - PowerShell: Set destination parameter for Get-ChocolateyUnzip to be required - to match what is used in Chocolatey CLI.
- Package Builder:
  - Fix - Ensure "--version" option is ignored when using choco new command- see [licensed #203](https://github.com/chocolatey/chocolatey-licensed-issues/issues/203).
  - Fix - Ensure checksums are calculated correctly when using both x86 and x64 urls that result in the same file name - see [licensed #94](https://github.com/chocolatey/chocolatey-licensed-issues/issues/94).
- Fix - Package Internalizer: Remove authenticode signatures when combining files - see [licensed #155](https://github.com/chocolatey/chocolatey-licensed-issues/issues/155).
- Fix - Package Builder UI: Ensure that long file paths don't wrap within input text boxes.

### Improvements

- Package Builder / Package Internalizer - Generated scripts should include validation of Chocolatey license - see [blog post](https://blog.chocolatey.org/2021/09/chocolatey-licensed-changes-restricted-to-licensed-nodes/) for additional information.
- PowerShell:

>   - Ensure that all Cmdlets are using consistent parameter names by using aliases.

Reviewer: ```suggestion
  - Ensure that all Chocolatey helper functions are using consistent parameter names by using aliases.
```

>   - Support UnzipLocation as an alias to the Destination parameter in the Get-ChocolateyUnzip function - matching with Chocolatey CLI v0.11.0 - see [licensed #243](https://github.com/chocolatey/chocolatey-licensed-issues/issues/243).

Reviewer: ```suggestion
  - Support `UnzipLocation` as an alias to the `Destination` parameter in the `Get-ChocolateyUnzip` function - matching with Chocolatey CLI v0.11.0 - see [licensed #243](https://github.com/chocolatey/chocolatey-licensed-issues/issues/243).
```

### Bug Fix

> - Fix - Filtering of sources doesn't return correct list when useBackgroundServiceWithSelfServiceSourcesOnly is disabled - see [licensed #263](https://github.com/chocolatey/chocolatey-licensed-issues/issues/263).

Reviewer: ```suggestion
- Fix - Filtering of sources doesn't return correct list when `useBackgroundServiceWithSelfServiceSourcesOnly` is disabled - see [licensed #263](https://github.com/chocolatey/chocolatey-licensed-issues/issues/263).
```

### Bug Fixes

- Self-Service / Background Mode:

>   - Fix - choco pin command does not work when running through Chocolatey Agent.
>   - Fix - Prevent invalid configuration value for commandExecutionTimeout.

Reviewer: ```suggestion
  - Fix - `choco pin` command does not work when running through Chocolatey Agent.
  - Fix - Prevent invalid configuration value for `commandExecutionTimeout`.
```

### Bug Fixes

- Self-Service / Background Mode:
  - Fix - Chocolatey GUI does not work with Self-Service in v2.1.0 - see [licensed #162](https://github.com/chocolatey/chocolatey-licensed-issues/issues/162).
- Fix - Chocolatey Licensed Extension not reporting version in configuration logs.

> - Fix - Warnings seen in Chocolatey Agent and Central Management about "Unable to set Chocolatey Version to Semantic version".

Reviewer: ```suggestion
- Fix - Warnings seen in Chocolatey Agent and Chocolatey Central Management about "Unable to set Chocolatey Version to Semantic version".
```

## 2.1.0 (June 18, 2020)

### Bug Fixes

- Self-Service / Background Mode:
  - Fix - List/Search command fails to produce results when going through background service - see [licensed #141](https://github.com/chocolatey/chocolatey-licensed-issues/issues/141).

>   - Fix - Attempts to configure background mode while loading configuration in Chocolatey Agent and Central Management.

Reviewer: ```suggestion
  - Fix - Attempts to configure background mode while loading configuration in Chocolatey Agent and Chocolatey Central Management.
```

  - Fix - Spaces are not handled properly in file path for generic scanner - see [licensed #135](https://github.com/chocolatey/chocolatey-licensed-issues/issues/135).
- Package Builder (Choco New):
  - Fix - Illegal characters in Support URL cause package not to be created - see [licensed #111](https://github.com/chocolatey/chocolatey-licensed-issues/issues/111).
- Package CDN:
  - Fix - Files on CDN with a '+' in the name are not downloaded - see [licensed #134](https://github.com/chocolatey/chocolatey-licensed-issues/issues/134).

### Improvements

- Package Internalizer (Choco Download):
  - Enhance `--no-progress` to not display download counters - see [licensed #104](https://github.com/chocolatey/chocolatey-licensed-issues/issues/104).
  - [Security] Files downloaded and internalized should be run against the virus scanner (if configured) - see [licensed #108](https://github.com/chocolatey/chocolatey-licensed-issues/issues/108).
- Disable existing context menus on upgrade when passing `/NoContextMenu` in package parameters.
- Remove context menus added during installation when uninstalling extension.

> - Central Management Deployments configuration additions.

Reviewer: ```suggestion
- Chocolatey Central Management Deployments configuration additions.
```

## 2.0.3 (January 30, 2020)

### Bug Fixes

- Chocolatey Central Management:
  - Fix - Add configuration parameter for size of message accepted by service - see [licensed #95](https://github.com/chocolatey/chocolatey-licensed-issues/issues/95).
- Self-Service / Background Mode:

>   - Fix - Arguments passed to background from GUI should be in same format as from direct choco.

Reviewer: ```suggestion
  - Fix - Arguments passed to background from GUI should be in same format as from direct Chocolatey CLI.
```

- Fix - Error when using {} in package parameters or silent args - see [licensed #88](https://github.com/chocolatey/chocolatey-licensed-issues/issues/88).
- Package Internalizer (Choco Download):

>   - Fix - Message regarding location of internalized nupkg is wrong when using output directory.

Reviewer: ```suggestion
  - Fix - Message regarding location of internalized package is wrong when using output directory.
```

### Improvements

- Package Builder UI / Package Upload UI - Authenticode Sign generated assemblies - see [licensed #116](https://github.com/chocolatey/chocolatey-licensed-issues/issues/116).

> - Package Internalizer (Choco Download) - Find Packages based on Repository Optimizations.

Reviewer: ```suggestion
- Package Internalizer - Find Packages based on Repository Optimizations.
```

## 2.0.2 (April 29, 2019)

### Bug Fix

> - Fix - Self-Service: Administrator being subjected to user-limited uninstalls when background service does not apply. See [Licensed #77](https://github.com/chocolatey/chocolatey-licensed-issues/issues/77).

Reviewer: ```suggestion
- Fix - Self-Service - Administrator being subjected to user-limited uninstalls when background service does not apply. See [Licensed #77](https://github.com/chocolatey/chocolatey-licensed-issues/issues/77).
```

## 2.0.1 (April 10, 2019)

### Bug Fixes

> - Fix - PowerShell - Licensed Install-ChocolateyInstallPackage doesn't work with Install Arguments that have apostrophes in the same way as open source - see [licensed #74](https://github.com/chocolatey/chocolatey-licensed-issues/issues/74).

Reviewer: ```suggestion
- Fix - PowerShell - Licensed `Install-ChocolateyInstallPackage` doesn't work with install arguments that have apostrophes in the same way as open source - see [licensed #74](https://github.com/chocolatey/chocolatey-licensed-issues/issues/74).
```

- Fix - Non 7-zip decompression method doesn't appear to work on newer Windows environments.

> - Fix - Add alias for old name of Install-AppSettingsJsonFile for compatibility - see [licensed #71](https://github.com/chocolatey/chocolatey-licensed-issues/issues/71).

Reviewer: ```suggestion
- Fix - Add alias for old name of `Install-AppSettingsJsonFile` for compatibility - see [licensed #71](https://github.com/chocolatey/chocolatey-licensed-issues/issues/71).
```

- Self-Service / Background Mode:
  - Fix - Incorrect handling in username does not allow Self-Service users to uninstall based on extra character present - see [licensed #73](https://github.com/chocolatey/chocolatey-licensed-issues/issues/73).
- Fix - Package Synchronizer - Show All Packages In Programs and Features Out of Preview.

## 2.0.0 (March 14, 2019)

### Features

- Chocolatey Central Management tie in - Brings Central Management client configuration to Chocolatey. Requires Chocolatey v0.10.12 along with Chocolatey Agent v0.9.0 to allow communication with Central Management.
- [Security] Self-Service - allow uninstalls of ONLY packages installed by a user. Allows restricting what a user can uninstall to only what they have installed - see [licensed #2](https://github.com/chocolatey/chocolatey-licensed-issues/issues/2).

### Bug Fixes

- Self-Service / Background Mode:

>   - Fix - Package arguments not being passed from Agent to choco.exe properly (client side fixes) - see [licensed #60](https://github.com/chocolatey/chocolatey-licensed-issues/issues/60).

Reviewer: ```suggestion
  - Fix - Package arguments not being passed from Chocolatey Agent to Chocolatey CLI properly (client side fixes) - see [licensed #60](https://github.com/chocolatey/chocolatey-licensed-issues/issues/60).
```

  - Fix - Background Service does not pass the exit code back to the console (client side) - see [licensed #51](https://github.com/chocolatey/chocolatey-licensed-issues/issues/51).
- Package Internalizer (Choco Download):

>   - Fix - PackageInternalizer - UseOriginalLocation switch being appended inappropriately when line continuation is used with no arguments on first line - see [licensed #64](https://github.com/chocolatey/chocolatey-licensed-issues/issues/64).

Reviewer: ```suggestion
  - Fix - PackageInternalizer - `UseOriginalLocation` option being appended inappropriately when line continuation is used with no arguments on first line - see [licensed #64](https://github.com/chocolatey/chocolatey-licensed-issues/issues/64).
```

- Package Upload UI:
  - Fix - Package Uploader is unable to push to a non https source - see [licensed #49](https://github.com/chocolatey/chocolatey-licensed-issues/issues/49).
- Package Builder (Choco New):
  - Fix - Nuspec files section should not be commented out as it doesn't allow subdirectories.
  - Fix - Ensure Architect and MSP are able to use Package Builder.
  - Fix - Escape XML characters in generated nuspec file.
- Windows Service Management Functions:
  - Fix - When shutting down a Windows service it may hold a lock on a file for a second after it shuts down.
- Fix - Type detection holds file locks if it times out - see [licensed #59](https://github.com/chocolatey/chocolatey-licensed-issues/issues/59).
- Fix - Ensure Architect edition has all features set appropriately.

### Improvements

- Package Builder - Remove "v" as part of "Name v2.2.23" In product names.
- Package Upload UI - Ability to auto populate Source dropdown list - see [licensed #50](https://github.com/chocolatey/chocolatey-licensed-issues/issues/50).
- Package Internalizer - Allow downloading all installed packages with `--installed-packages`.

> - PowerShell - Cmdlets for Getting/Setting encrypted values in JSON files.

Reviewer: ```suggestion
- PowerShell - Functions for Getting/Setting encrypted values in JSON files.
```

## 1.12.12 (June 12, 2018)

### Features

> - Package Upload UI - Ability to right click on a nupkg, and access UI for selecting where package is pushed to.

Reviewer: ```suggestion
- Package Upload UI - Ability to right click on a package, and access UI for selecting where package is pushed to.
```

### Bug Fixes

> - Fix - choco new generates uninstall template with wrong use of registry key variable - see [#1304](https://github.com/chocolatey/choco/issues/1304).

Reviewer: ```suggestion
- Fix - `choco new` generates uninstall template with wrong use of registry key variable - see [#1304](https://github.com/chocolatey/choco/issues/1304).
```

- Package Internalizer (Choco Download):
  - Fix - Append UseOriginalLocation for multiline so that it adds a continuation and not a hanging parameter on a new line.
  - Fix - Variable replacement wasn't working for `${value}` and `$($value)`.
  - Fix - Find and replace the original url in the script.

### Improvements

- [Security] Allow locking down new/download commands to admin only - added as feature flip switches (see `choco feature list`).
- Self-Service / Background Mode:

>   - Provide the user context to Chocolatey on install / upgrade (user context is original user requesting action).

Reviewer: ```suggestion
  - Provide the user context to Chocolatey CLI on install / upgrade (user context is original user requesting action).
```

## 1.9.8 (March 25, 2017)

### Bug Fixes

- Fix - Ensure Chocolatey Licensed is compatible with Chocolatey v0.10.4.
- Fix - AutoUninstaller - ensure uninstallExe is split by quotes when necessary - see [#1208](https://github.com/chocolatey/choco/issues/1208).
- Package Synchronizer (Choco Sync):
  - Fix - Ensure template properties are cleared, even when skipping due to errors.
- Package Builder (Choco New):
  - Fix - Ensure authors is never empty.
  - Fix - Before setting properties from registry, ensure they have a value.
  - Fix - Replace invalid ".-" in package id.

## 1.9.7 (March 20, 2017)

### Bug Fixes

- Fix - Support automatic decompression on downloads - see [#1056](https://github.com/chocolatey/choco/issues/1056).
- Fix - Package Builder - Restrict Get-UninstallRegistryKey params in chocolateyUninstall.ps1.
- Fix - Package Internalizer - exit non-zero when variable replacement fails.

### Improvements

- Ensure PowerShell scripts use CRLF so authenticode verification doesn't fail.
- Install Directory should see `INSTALLFOLDER` MSI Property.
- Virus Scanner - provide context for other answers (No/Skip).

> - Allow both 32bit and 64bit file parameters with Install-ChocolateyInstallPackage - see [#1187](https://github.com/chocolatey/choco/issues/1187).

Reviewer: ```suggestion
- Allow both 32bit and 64bit file parameters with `Install-ChocolateyInstallPackage` - see [#1187](https://github.com/chocolatey/choco/issues/1187).
```

### Bug Fixes

- Fix - Error when running Install-ChocolateyInstallPackage without specifying silent arguments.

> - Remove *.istext file before the content-type check that creates file - see [#1012](https://github.com/chocolatey/choco/issues/1012).

Reviewer: ```suggestion
- Remove `*.istext` file before the content-type check that creates file - see [#1012](https://github.com/chocolatey/choco/issues/1012).
```

### Features

- Package Synchronizer v2 - `choco sync` to associate existing packages with unmanaged software - see [the documentation](https://docs.chocolatey.org/en-us/features/package-synchronization#sync-command).
- Package Builder UI:
  - Starts at Professional edition (minus autodetection).
  - Tabbed interface.
  - Tab for entire Nuspec.
  - [Business] Tab to generate from Programs and Features.

### Bug Fixes

- Self-Service / Background Mode:
  - Fix - Add a line after progress is complete.
- Fix - Automatic creation of ignore file in Install-ChocolateyInstallPackage throws errors when it fails - see [#380](https://github.com/chocolatey/chocolatey/issues/380) for original issue.
- Package Builder (Choco New):
  - Fix - remove "version/ver" if next to version number in DisplayName.
  - Fix - todo / logging need to escape curly braces to properly format.
  - Fix - continue on error.

### Improvements

- User can turn on Preview Features - `choco feature enable --name allowPreviewFeatures`.

> - InstallDirectory switch added to Install-ChocolateyInstallPackage.

Reviewer: ```suggestion
- InstallDirectory switch added to `Install-ChocolateyInstallPackage`.
```

- Package Internalizer (Choco Download):
  - option to ignore dependencies `--ignore-dependencies`.

> - Package Builder (Choco New):

Reviewer: ```suggestion
- Package Builder (`choco new`):
```

### Features

- Package Downloader/Internalizer (Choco Download):
  - Internalize package dependencies.
  - Allow download to separate location than where package will edit for internal resources (`--download-location`).

### Bug Fixes

- Fix - Directory Override - ensure override switch is passed to .install/.portable package when passing switch to meta package.
- Fix - Ensure web requests don't time out / are configurable - see  [#732](https://github.com/chocolatey/choco/issues/732).
- Package Builder (Choco New):
  - Fix - Cannot bind parameter because parameter 'fileType' is specified more than once for `--keep-remote`.
  - Fix - UNC paths with `--use-original-location` should not use `Install-ChocolateyPackage`.
  - Fix - Escape double quotes in PowerShell strings.
- Package Downloader/Internalizer (Choco Download):
  - Fix - Ignore commented out urls.
  - Fix - Do not download from a url more than once even if a package has it listed more than once.
  - Fix - Do not timeout for larger files.

### Improvements

- Directory Override - if MSI properties include INSTALLDIR or INSTALLLOCATION, use that instead of TARGETDIR.
- Package Builder (Choco New):

>   - MSI Properties generated are cleaned up, duplicates removed from chocolateyInstall.ps1 comments.

Reviewer: ```suggestion
  - MSI Properties generated are cleaned up, duplicates removed from `chocolateyInstall.ps1` comments.
```

  - Generate from Programs and Features is faster, does not repeat for Uninstaller keys in both 32bit/64bit registry hives.
  - Show downloaded file progress.
- Package Downloader/Internalizer (Choco Download):

>   - When chocolateyInstall.ps1 uses other ps1 files, ensure those are loaded for token replacement.

Reviewer: ```suggestion
  - When `chocolateyInstall.ps1` uses other ps1 files, ensure those are loaded for token replacement.
```

  - Show downloaded file progress.

## 1.8.2 (December 12, 2016)

### Bug Fixes

- Package Builder (Choco New):

>   - Fix - Ignore first argument for name if not a name. This will also be fixed in Chocolatey v0.10.4 with [#1085](https://github.com/chocolatey/choco/issues/1085).

Reviewer: ```suggestion
  - Fix - Ignore first argument for name if not a name. This will also be fixed in Chocolatey CLI v0.10.4 with [#1085](https://github.com/chocolatey/choco/issues/1085).
```

  - AutoUninstaller - determine type from original executable when FOSS is not able to detect installer type.
- Package Builder (Choco New):
  - Now supports downloading from url/url64 and determining whether to keep those files remote.
  - Switch to use original file location instead of copying into package.
  - Specify both 32-bit/64-bit file.
  - Work with zip files.

### Bug Fixes

- Fix - changes related to working directory fixes for [#937](https://github.com/chocolatey/choco/issues/937).
- Fix - double chocolatey folder name is not also applied to the passed in file name - see  [#908](https://github.com/chocolatey/choco/issues/908).
- Package Builder (Choco New):
  - Fix - remove parentheses from package names.
  - Fix - keep template updated.
- Package Internalizer (Choco Download):
  - Fix - handle downloaded files with the same name Sometimes the file name is the same when the architecture is different. Handle that by using the url counter for all additional files with the same name.

### Improvements

- Remind About Upcoming Expiration - when the license is expiring within a month's time, remind the user about renewal.
- Package Builder (Choco New):

>   - Handle -forcex86 with package creation.

Reviewer: ```suggestion
  - Handle `-forcex86` with package creation.
```

  - Add shimgen ignore for exes.
  - Use ProductVersion when version 0.0.0.0.
  - Remove the word "installer" from package name.
  - Allow specifying name of the package.
  - Allow template override with warning.
  - Show MSI properties in install script (commented).
  - Ensure `ALLUSERS=1` when an MSI is set to per user by default.
  - Automatically checksum files.
  - Allow files to stay remote - use remote helpers when files stay remote.
- Package Internalizer (Choco Download):
  - handle variables in urls set like ${word}.

>   - Append `-UseOriginalLocation` to the end of the arguments passed to Install-ChocolateyPackage. Work with splatting properly as well.

Reviewer: ```suggestion
  - Append `-UseOriginalLocation` to the end of the arguments passed to `Install-ChocolateyPackage`. Work with splatting properly as well.
```

## 1.5.1 (Aug 9, 2016)

### Bug Fix

- Fix - Valid Exit Codes do not support values bigger than Int32.MaxValue. - see [#900](https://github.com/chocolatey/choco/issues/900).

### Improvement

> - Package Internalizer (Choco Download) - specify resources location (when not embedding into package).

Reviewer: ```suggestion
- Package Internalizer (`choco download`) - specify resources location (when not embedding into package).
```

- BETA Testers - Recompiled packages support - Download a package and all remote resources, recompiling the package to use local resources instead.
- BETA Testers - Synchronize w/Programs and Features - Chocolatey synchronizes manually uninstalled software with package state.
- [Business] Create Packages from Installers aka Package Builder! Create packages directly from software installers in seconds! **Chocolatey for Business can automatically create packages for all the software your organization uses in under 5 minutes!**

> - New Command! choco support - quickly see how you can contact support - [#745](https://github.com/chocolatey/choco/issues/745).

Reviewer: ```suggestion
- New Command! `choco support` - quickly see how you can contact support - [#745](https://github.com/chocolatey/choco/issues/745).
```
